[java] automatic-module-name header added for pmd-core

magwas opened the pull request:
<!-- Please, prefix the PR title with the language it applies to within brackets, such as [java] or [apex] -->

## Describe the PR

Adds Automatic-Module-Name header to the MANIFEST.MF of pmd-core
Had to exclude pmd.cache from javadoc, as there are no classes there, but checked that subpackages get included in the javadoc.
## Related issues

<!-- PR relates to issues in the `pmd` repo: -->

- Fix #6167
- Related #4832

## Ready?

<!-- If you feel like you can help to check off the following tasks, that'd be great. If not, don't worry - we will take care of it. -->

- [ ] Added unit tests for fixed bug/feature
- [x ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

magwas committed:
automatic-module-name header added for pmd-core

magwas committed:
automatic module name for pmd-java

adangel commented on pmd-core/pom.xml:
```diff
@@ -74,6 +88,7 @@
                 <configuration>
                     <!-- overrides the configuration from parent pom: we have here no offline links yet -->
                     <offlineLinks combine.self="override"></offlineLinks>
+                    <excludePackageNames>net.sourceforge.pmd.cache</excludePackageNames>
```
> Had to exclude pmd.cache from javadoc, as there are no classes there, but checked that subpackages get included in the javadoc.

That's strange, maybe a bug in javadoc? We have at least a package-info.java file there, so we should not run into such a problem, should we?

magwas committed:
draft changes for adr-3

magwas committed:
added automatic-module-name-header for most modules

scala and visualforce uses artifact naming conventions clashing with the module system,
and their ecosystem does not have module naming convention, so they got no header for now
